charts/consul/templates/cni-serviceaccount.yaml

```diff
@@ -0,0 +1,14 @@
+{{- if (and (.Values.connectInject.cni.enabled) (not .Values.connectInject.transparentProxy.defaultEnabled)) }}{{ fail "connectInject.transparentProxy.defaultEnabled must be true if connectInject.cni.enabled is true" }}{{ end -}}
+{{- if (and .Values.connectInject.cni.enabled .Values.connectInject.transparentProxy.defaultEnabled) }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "consul.fullname" . }}-cni
+  namespace: {{ .Values.connectInject.cni.namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: cni 
```

Comment:
need to add support for `imagePullSecrets`: https://github.com/hashicorp/consul-k8s/blob/023007f6b830a08df876557a2cd858120eb46975/charts/consul/templates/connect-inject-serviceaccount.yaml#L17-L21

Reply:
Thanks!  (and done)